--- a/dma.c
+++ b/dma.c
@@ -1169,7 +1169,8 @@ void mt76_dma_cleanup(struct mt76_dev *dev)
 	int i;
 
 	mt76_worker_disable(&dev->tx_worker);
-	napi_disable(&dev->tx_napi);
+	if (test_bit(NAPI_STATE_LISTED, &dev->tx_napi.state))
+		napi_disable(&dev->tx_napi);
 	netif_napi_del(&dev->tx_napi);
 
 	for (i = 0; i < ARRAY_SIZE(dev->phys); i++) {
@@ -1189,6 +1190,8 @@ void mt76_dma_cleanup(struct mt76_dev *dev)
 	mt76_for_each_q_rx(dev, i) {
 		struct mt76_queue *q = &dev->q_rx[i];
 
+		if (test_bit(NAPI_STATE_LISTED, &dev->napi[i].state))
+			napi_disable(&dev->napi[i]);
 		netif_napi_del(&dev->napi[i]);
 		mt76_dma_rx_cleanup(dev, q);
 
