--- a/loginutils/chpasswd.c
+++ b/loginutils/chpasswd.c
@@ -110,8 +110,11 @@
 		logmode = LOGMODE_BOTH;
 		if (rc < 0)
 			bb_error_msg_and_die("an error occurred updating password for %s", name);
-		if (rc)
-			bb_info_msg("password for '%s' changed", name);
+
+		/* comment by dimmalex for no log interference */
+		//if (rc)
+		//	bb_info_msg("password for '%s' changed", name);
+
 		logmode = LOGMODE_STDIO;
 		free(name);
 		free(free_me);
